app/assets/javascripts/controllers/project_roles_ctrl.js

```diff
@@ -0,0 +1,69 @@
+samson.controller('ProjectRolesCtrl', function($scope, $element, $filter, userProjectRoleFactory, projectRoleFactory, projectRolesService, messageCenterService) {
+  $scope.project_role = {};
+  $scope.roles = [];
+
+  $scope.initModel = function() {
+    $scope.project_role = userProjectRoleFactory.buildFromDom($element[0]);
+    loadProjectRoles();
+  };
+
+  $scope.$watch('project_role.role_id', function(new_role_value, old_role_value) {
```

Comment:
To be honest, I didn't even think about the performance side of things, as this Angular app is so small and so simple, but you're totally right. I threw it into a watch expression in the controller to keep the HTML simpler, instead of having to put the ng-change expression in each radio button, but there's no point on overloading the digest cycle with this.

I'm going to change it. Thanks for the heads up.